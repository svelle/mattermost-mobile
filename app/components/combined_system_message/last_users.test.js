// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See License.txt for license information.

import React from 'react';
import {configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({adapter: new Adapter()});

import {Posts} from 'mattermost-redux/constants';

import {shallowWithIntl} from 'test/intl-test-helper';

import LastUsers from './last_users';

describe('LastUsers', () => {
    const baseProps = {
        actor: 'actor',
        expandedLocale: {id: 'expanded_locale_id', defaultMessage: 'Expanded Locale'},
        postType: Posts.POST_TYPES.ADD_TO_TEAM,
        style: {activityType: {fontSize: 14}, link: 1},
        userDisplayNames: ['User One', 'User Two'],
    };

    test('should match snapshot', () => {
        const wrapper = shallowWithIntl(
            <LastUsers {...baseProps}/>
        );

        const expanded = wrapper.instance().renderExpandedView(baseProps.expandedLocale, baseProps.userDisplayNames, baseProps.actor, 1, baseProps.style);
        expect(expanded).toMatchSnapshot();

        const collapsed = wrapper.instance().renderCollapsedView(baseProps.postType, baseProps.userDisplayNames, baseProps.actor, 1, baseProps.style);
        expect(collapsed).toMatchSnapshot();
    });

    test('should match state on handleOnClick', () => {
        const wrapper = shallowWithIntl(
            <LastUsers {...baseProps}/>
        );

        wrapper.setState({expand: false});
        wrapper.instance().handleOnClick({preventDefault: jest.fn()});
        expect(wrapper.state('expand')).toEqual(true);
    });
});
