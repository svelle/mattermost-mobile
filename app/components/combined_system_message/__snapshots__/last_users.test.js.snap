// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LastUsers should match snapshot 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
>
  <InjectIntl(FormattedText)
    defaultMessage="E"
    id="e"
    values={
      Object {
        "lastUser": "User Two",
        "users": "User One",
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 14,
      }
    }
  >
    <InjectIntl(FormattedText)
      defaultMessage="x"
      id="x"
    />
  </Text>
  <InjectIntl(FormattedText)
    defaultMessage="p"
    id="p"
    values={
      Object {
        "actor": "actor",
      }
    }
  />
</Text>
`;

exports[`LastUsers should match snapshot 2`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
>
  <InjectIntl(FormattedText)
    defaultMessage="{firstUser} and "
    id="mobile.combined_system_message.first_user_and"
    values={
      Object {
        "firstUser": "User One",
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={1}
  >
    <InjectIntl(FormattedText)
      defaultMessage="{numOthers} others "
      id="mobile.combined_system_message.others"
      values={
        Object {
          "numOthers": 1,
        }
      }
    />
  </Text>
  <InjectIntl(FormattedText)
    defaultMessage="were "
    id="mobile.combined_system_message.were"
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 14,
      }
    }
  >
    <InjectIntl(FormattedText)
      defaultMessage="added to the team"
      id="mobile.combined_system_message.added_to_team"
    />
  </Text>
  <InjectIntl(FormattedText)
    defaultMessage=" by {actor}."
    id="mobile.combined_system_message.by_actor"
    values={
      Object {
        "actor": "actor",
      }
    }
  />
</Text>
`;
